### PR TITLE
feat: adding unique index to the slug field in tenant collection.

### PR DESCRIPTION
**Pull request by Naveen-Goud**
PR for **Task - #33377**  adding unique index to the slug field in tenant collection.
 
 **What I have done in this PR:**
 
 1.Updated the **@unique** with **@Indexed(Unique=true)** in the tenant collection.
 2.Added a changelog file for the migration of the data.
 3.Migration Step also  checks the slug ids are unique, and trying to insert with the same data also  returns a error message.
 
@nidhi-nair 
@NilanshBansal
 I have created this pr for the refernce and also would like to get clarification on the question I have asked.

 thank you.

### CODE REVIEW

**Review by nsarupr — CHANGES_REQUESTED** [review]
HI @Naveen-Goud , thanks for taking out time to make contributions to `appsmith_`. I have reviewed the PR. Please go through the changes requested.

**Review by NilanshBansal — APPROVED** [review]
LGTM

**Review by NilanshBansal — CHANGES_REQUESTED** [review]
@Naveen-Goud can you please use makeIndex() method instead of creating index name.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Tenant.java [hunk 1]
```diff
@@ -18,7 +18,8 @@
 @FieldNameConstants
 public class Tenant extends BaseDomain {
 
-    @Unique String slug;
+    @Indexed(Unique= true)
```
Comment (NilanshBansal) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Tenant.java:21]: @Naveen-Goud this change is not required, only migration should be sufficient.
Also Unique = true returns a compilation error.

**Comment by NilanshBansal** [comment]
@Naveen-Goud we might not need to implement this right now because we haven't yet moved to multi-tenancy fully. 
There is always one document in the database.
image
Ref:https://github.com/appsmithorg/appsmith/issues/33377#issuecomment-2144763283

Apologies for the inconvenience, thank you for all your efforts in contributing to appsmith. You can continue contributing to appsmith by selecting issues from [this](https://github.com/appsmithorg/appsmith/issues?q=is%3Aopen+is%3Aissue+label%3A%22Good+First+Issue%22) list. Looking forward to your valuable contributions 🙏

**Pull request closed without merging**